src/main/java/app/gpx_animator/Renderer.java

```diff
@@ -722,14 +723,17 @@ private Point2D drawMarker(final BufferedImage bi, final int frame) {
                     g2.setColor(ceilingEntry == null ? Color.white : trackConfiguration.getColor());
 
                     final TrackIcon trackIcon = trackConfiguration.getTrackIcon();
-                    if (trackIcon == null || trackIcon.getKey().isEmpty()) {
-                        drawSimpleCircleOnGraphics2D(point, g2);
-                    } else {
-                        try {
-                            drawIconOnGraphics2D(point, g2, trackIcon);
-                        } catch (final IOException e) {
+                    final File trackIconFile = trackConfiguration.getInputIcon();
+                    try {
+                        if (trackIconFile != null && trackIconFile.exists()) {
+                            drawIconFileOnGraphics2D(point, g2, trackIconFile, trackConfiguration.getFlipIcon());
+                        } else if (trackIcon != null && !trackIcon.getKey().isEmpty()) {
+                            drawIconOnGraphics2D(point, g2, trackIcon, trackConfiguration.getFlipIcon());
+                        } else {
                             drawSimpleCircleOnGraphics2D(point, g2);
                         }
+                    } catch (final IOException e) {
+                        drawSimpleCircleOnGraphics2D(point, g2);
```

Comment:
A fallback in case of an exception is good, but it could result in unexpected behavior and the user did not know, why. The user selects an icon but gets just a circle without a notification. I would suggest throwing an `UserException` with a descriptive message (with a translation, so use the `resourceBundle` like in other messages). The message of an `UserException` is shown to the user, so he can react and check his settings.

Reply:
Agreed, refactored on my master